app/Jobs/ProcessMediaWikiJobsJob.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace App\Jobs;
+
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Contracts\Queue\ShouldBeUnique;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Support\Facades\Log;
+use Maclof\Kubernetes\Client;
+use Maclof\Kubernetes\Models\Job as KubernetesJob;
+
+class ProcessMediaWikiJobsJob implements ShouldQueue, ShouldBeUnique
+{
+    use InteractsWithQueue, Queueable;
+
+    private string $wikiDomain;
+    private string $jobsKubernetesNamespace;
+    private int $apiJobConcurrencyLimit;
+
+    public function __construct (string $wikiDomain)
+    {
+        $this->wikiDomain = $wikiDomain;
+        $this->jobsKubernetesNamespace = env('API_JOB_NAMESPACE', 'api-jobs');
+    }
+
+    public function uniqueId(): string
+    {
+        return $this->wikiDomain;
+    }
+
+    public function handle (Client $kubernetesClient): void
+    {
+        $filterCompletedJobs = function (KubernetesJob $job, int $key): bool {
+            $phase = data_get($job->toArray(), 'status.conditions.0.type', null);
```

Comment:
This only seems to work for jobs that are backed by a pod. Jobs that failed creation because they could not be scheduled report `{}` as their status. Not entirely sure right now how to best match these.